doc/plugin_tutorial/tuto1/src/simple_declare.ml

```diff
@@ -9,16 +9,17 @@ let edeclare ident (_, poly, _ as k) ~opaque sigma udecl body tyopt imps hook =
      (Option.List.cons tyopt [body]) in
   let sigma = Evd.restrict_universe_context sigma uvars in
   let univs = Evd.check_univ_decl ~poly sigma udecl in
+  let uctx = Evd.evar_universe_context sigma in
   let ubinders = Evd.universe_binders sigma in
   let ce = Declare.definition_entry ?types:tyopt ~univs body in
-  DeclareDef.declare_definition ident k ce ubinders imps ~hook
+  DeclareDef.declare_definition ident k ce ubinders imps ~hook_data:(hook,uctx,[])
 
 let packed_declare_definition ~poly ident value_with_constraints =
   let body, ctx = value_with_constraints in
   let sigma = Evd.from_ctx ctx in
   let k = (Decl_kinds.Global, poly, Decl_kinds.Definition) in
   let udecl = UState.default_univ_decl in
-  let nohook = Lemmas.mk_hook (fun _ x -> ()) in
+  let nohook = Lemmas.mk_hook (fun _ _ _ x -> ()) in
```

Comment:
Should we keep messing around with empty hooks in the plugin tutorial? I'm inclined to just remove the hook argument to edeclare.

Reply:
I was about to make it optional, but I hesitated due to the fact that people may forget them. As you wish.

Reply:
Anyways if everything goes well we will greatly refactor hooks / terminators so this won't stay for long.